tests/helpers/ast_helpers.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# ======================================================================================================================
+# Imports
+# ======================================================================================================================
+import ast
+
+
+class BaseDecoratorRetriever(ast.NodeVisitor):
+
+    def _generate_pytest_decorators(self, decorators):
```

Comment:
An improvement to this would probably be to just read the lines from disk rather than trying to reconstruct them from the AST.  🤷‍♂️

Reply:
This is an improvement that we can make if we merge this branch into master.